File: epic-games-hub/src/main/java/com/epicgameshub/epic_games_hub/game/GameResponse.java
package com.epicgameshub.epic_games_hub.game;

import com.epicgameshub.epic_games_hub.feedback.Feedback;
import com.epicgameshub.epic_games_hub.history.GameTransactionHistory;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class GameResponse {
    private Integer id;
    private String title;
    private String genre;
    private String platform;
    private String publisher;
    private String releaseDate;
    private String description;
    private double rating;
    private boolean availability;
    private boolean sharable;
    private byte[] coverImage;
    private String owner;
    private List<Feedback> feedbacks;
    private List<GameTransactionHistory> histories;
}

File: epic-games-hub-ui/src/app/services/models/game-response.ts
/* tslint:disable */
/* eslint-disable */
import { Feedback } from '../models/feedback';
import { GameTransactionHistory } from '../models/game-transaction-history';
export interface GameResponse {
  availability?: boolean;
  coverImage?: Array<string>;
  description?: string;
  feedbacks?: Array<Feedback>;
  genre?: string;
  histories?: Array<GameTransactionHistory>;
  id?: number;
  owner?: string;
  platform?: string;
  publisher?: string;
  rating?: number;
  releaseDate?: string;
  sharable?: boolean;
  title?: string;
}
